app/controllers/passwords_controller.rb

```diff
@@ -23,45 +27,41 @@ def edit
     else
       # When user doesn't have mfa, a valid token is a full "magic link" sign in.
       verified_sign_in
-      render template: "passwords/edit"
+      render :edit
+    end
+  end
+
+  def create
+    user = User.find_by_normalized_email(@email)
+
+    if user
+      user.forgot_password!
+      ::PasswordMailer.change_password(user).deliver_later
     end
+
+    render :create, status: :accepted
```

Comment:
This lands you on a page that 404s if you hit enter on the url bar, and asks about form resubmit if you reload. I think it'd be better off going back to the signin page with a flash message set. What do you think?